src/frontend/src/lib/components/transactions/Transaction.svelte

```diff
@@ -126,14 +123,53 @@
 					{/if}
 				{/if}
 			{/snippet}
+			{#snippet amountDescription()}
+				{#if nonNullish(timestamp)}
+					<span data-tid="receive-tokens-modal-transaction-timestamp">
+						{formatSecondsToDate({
+							seconds: Number(timestamp),
+							language: $currentLanguage,
+							formatOptions: {
+								hour: '2-digit',
+								minute: '2-digit',
+								hour12: false
+							}
+						})}
+					</span>
+				{/if}
+			{/snippet}
 
 			{#snippet description()}
-				<span data-tid="receive-tokens-modal-transaction-timestamp">
-					{#if nonNullish(timestamp)}
-						{formatSecondsToDate({ seconds: Number(timestamp), language: $currentLanguage })}
+				<span class="inline-flex min-w-0 items-center gap-2 text-primary">
+					{#if type === 'send'}
+						<span class="shrink-0">{$i18n.transaction.text.to}</span>
+					{:else if type === 'receive'}
+						<span class="shrink-0">{$i18n.transaction.text.from}</span>
+					{/if}
+
+					{#if nonNullish(contact)}
+						<span class="shrink-0">
+							<Avatar name={contact.name} image={contact.image} variant="xxs" />
+						</span>
 					{/if}
+
+					<span class="inline-flex min-w-0 items-center gap-1">
+						<span>
+							{#if nonNullish(contact)}
+								{contact.name}
+							{:else if nonNullish(contactAddress)}
+								{shortenWithMiddleEllipsis({ text: contactAddress })}
+							{/if}
+						</span>
+						{#if nonNullish(addressAlias) && addressAlias !== ''}
```

Comment:
you can use the util `notEmptyString ` that does both these checks
```suggestion
						{#if notEmptyString(addressAlias)}
```